posthog/temporal/workflows/postgres_batch_export.py

```diff
@@ -0,0 +1,264 @@
+import contextlib
+import datetime as dt
+import json
+from dataclasses import dataclass
+
+import psycopg2
+from django.conf import settings
+from psycopg2 import sql
+from temporalio import activity, workflow
+from temporalio.common import RetryPolicy
+
+from posthog.batch_exports.service import PostgresBatchExportInputs
+from posthog.temporal.workflows.base import (
+    CreateBatchExportRunInputs,
+    PostHogWorkflow,
+    UpdateBatchExportRunStatusInputs,
+    create_export_run,
+    update_export_run_status,
+)
+from posthog.temporal.workflows.batch_exports import (
+    BatchExportTemporaryFile,
+    get_data_interval,
+    get_results_iterator,
+    get_rows_count,
+)
+from posthog.temporal.workflows.clickhouse import get_client
+
+
+@contextlib.contextmanager
+def postgres_connection(inputs):
+    """Manage a Postgres connection."""
+    connection = psycopg2.connect(
+        user=inputs.user,
+        password=inputs.password,
+        database=inputs.database,
+        host=inputs.host,
+        port=inputs.port,
+    )
+
+    try:
+        yield connection
+    except Exception:
+        connection.rollback()
+        raise
+    else:
+        connection.commit()
+    finally:
+        connection.close()
+
+
+def copy_tsv_to_posgtres(tsv_file, postgres_connection, schema: str, table_name: str, schema_columns):
```

Comment:
nit: typo
```suggestion
def copy_tsv_to_postgres(tsv_file, postgres_connection, schema: str, table_name: str, schema_columns):
```

Reply:
Haha, let me make a proper search and replace just in case I copied this name somewhere else.